### PR TITLE
remove duplicate comments in EncryptedEnv

### PR DESCRIPTION
**Pull request by acelyc111**
There are some comments on subclasses in EncryptedEnv module which are duplicate to their parent classes, it would be nice to remove the duplication and keep the consistency if the comments on parent classes updated in someday.

### CODE REVIEW

File: include/rocksdb/env_encryption.h [hunk 1]
```diff
@@ -23,9 +23,6 @@ struct ConfigOptions;
 // read from disk.
 Env* NewEncryptedEnv(Env* base_env,
                      const std::shared_ptr<EncryptionProvider>& provider);
-std::shared_ptr<FileSystem> NewEncryptedFS(
-    const std::shared_ptr<FileSystem>& base_fs,
-    const std::shared_ptr<EncryptionProvider>& provider);
```
Comment (ajkr) [inline include/rocksdb/env_encryption.h:28]: Why delete this?

**Review by ajkr — APPROVED** [review]
LGTM, thanks!

File: include/rocksdb/env_encryption.h (hunk 1 above)
Comment (acelyc111) [reply include/rocksdb/env_encryption.h]: I saw it was only used in NewEncryptedEnv in .cc file, thought nobody will use it.
But consider it is in `include/*` directory, I've reverted this change.

Comment (ajkr) [reply include/rocksdb/env_encryption.h]: Thanks!

**Pull request closed without merging**